homeassistant/components/zerproc/light.py

```diff
@@ -0,0 +1,194 @@
+        try:
+            await discover_entities(hass, async_add_entities)
+            warned = False
+        except pyzerproc.ZerprocException:
+            if warned is False:
+                _LOGGER.warning("Error discovering Zerproc lights", exc_info=True)
+                warned = True
+
+    # Initial discovery
+    hass.async_add_job(discover)
+
+    # Perform recurring discovery of new devices
+    async_track_time_interval(hass, discover, DISCOVERY_INTERVAL)
+
+
+class ZerprocLight(Light):
+    """Representation of an Zerproc Light."""
+
+    def __init__(self, hass, light):
+        """Initialize a Zerproc light."""
+        self._light = light
+        self._name = None
+        self._is_on = None
+        self._hs_color = None
+        self._brightness = None
+        self._available = True
+
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.on_hass_shutdown)
+
+    def on_hass_shutdown(self, event):
+        """Execute when Home Assistant is shutting down."""
+        self._light.disconnect()
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._light.name
+
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return self._light.address
+
+    @property
+    def device_info(self):
+        """Device info for this light."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+        }
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_ZERPROC
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the hs color."""
+        return self._hs_color
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._is_on
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        if ATTR_BRIGHTNESS in kwargs or ATTR_HS_COLOR in kwargs:
+            default_hs = (0, 0) if self._hs_color is None else self._hs_color
+            hue_sat = kwargs.get(ATTR_HS_COLOR, default_hs)
+
+            default_brightness = 255 if self._brightness is None else self._brightness
+            brightness = kwargs.get(ATTR_BRIGHTNESS, default_brightness)
+
+            rgb = color_util.color_hsv_to_RGB(*hue_sat, brightness / 255 * 100)
+            self._light.set_color(*rgb)
+        else:
+            self._light.turn_on()
+
+    def turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        self._light.turn_off()
+
+    def update(self):
+        """Fetch new state data for this light."""
+        try:
+            state = self._light.get_state()
```

Comment:
I am assuming that each light has their own IP. In that case set `PARALLEL_UPDATES = 0` on the top of the file and Home Assistant will talk to each light in parallel.